### ISSUE TITLE
FRR exporter reports OSPF collector down after FRR service restart

### ISSUE THREAD

**Issue by karlism** [tynany/frr_exporter#23]
There is an issue with FRR exporter after FRR service is restarted, it reports that OSPF collector is down, FRR exporter restart resolves this problem:
```
# curl -s localhost:9342/metrics | grep frr_collector_up
# HELP frr_collector_up Whether the collector's last scrape was successful (1 = successful, 0 = unsuccessful).
# TYPE frr_collector_up gauge
frr_collector_up{collector="bgp"} 1
frr_collector_up{collector="ospf"} 1

# systemctl stop frr

# curl -s localhost:9342/metrics | grep frr_collector_up
# HELP frr_collector_up Whether the collector's last scrape was successful (1 = successful, 0 = unsuccessful).
# TYPE frr_collector_up gauge
frr_collector_up{collector="bgp"} 0
frr_collector_up{collector="ospf"} 0

# systemctl start frr

# curl -s localhost:9342/metrics | grep frr_collector_up
# HELP frr_collector_up Whether the collector's last scrape was successful (1 = successful, 0 = unsuccessful).
# TYPE frr_collector_up gauge
frr_collector_up{collector="bgp"} 1
frr_collector_up{collector="ospf"} 0

# systemctl restart frr_exporter

# curl -s localhost:9342/metrics | grep frr_collector_up
# HELP frr_collector_up Whether the collector's last scrape was successful (1 = successful, 0 = unsuccessful).
# TYPE frr_collector_up gauge
frr_collector_up{collector="bgp"} 1
frr_collector_up{collector="ospf"} 1
```

FRR exporter is started with following flags: `frr_exporter --collector.bgp.peer-descriptions --collector.bgp.peer-descriptions.plain-text --no-collector.bfd`

Log output during these queries is following:
```
$ frr_exporter --collector.bgp.peer-descriptions --collector.bgp.peer-descriptions.plain-text --no-collector.bfd
INFO[0000] Starting frr_exporter (version=0.2.11, branch=HEAD, revision=7a774985ce8463144f534417d1761ed7b3ff4e19) on :9342  source="frr_exporter.go:118"
ERRO[0063] collector "ospf" scrape failed: cannot get ospf interface summary: exit status 1  source="collector.go:138"
ERRO[0063] collector "bgp" scrape failed: cannot get bgp ipv4 unicast summary: exit status 1  source="collector.go:138"
ERRO[0073] collector "ospf" scrape failed: cannot get ospf interface summary: exit status 1  source="collector.go:138"
ERRO[0079] collector "ospf" scrape failed: cannot get ospf interface summary: exit status 1  source="collector.go:138"
^C

$ frr_exporter --collector.bgp.peer-descriptions --collector.bgp.peer-descriptions.plain-text --no-collector.bfd
INFO[0000] Starting frr_exporter (version=0.2.11, branch=HEAD, revision=7a774985ce8463144f534417d1761ed7b3ff4e19) on :9342  source="frr_exporter.go:118"
```

**Comment by tynany** [comment]
I am unable to replicate this behaviour. Are you able to share your `frr.conf` and the version of FRR you are using?

**Comment by karlism** [comment]
Thnaks for looking into this! FRR is version 7.4:
```
# /usr/lib/frr/ospfd -v
ospfd version 7.4
Copyright 1996-2005 Kunihiro Ishiguro, et al.
configured with:
	'--build=x86_64-redhat-linux-gnu' '--host=x86_64-redhat-linux-gnu' '--program-prefix=' '--disable-dependency-tracking' '--prefix=/usr' '--exec-prefix=/usr' '--bindir=/usr/bin' '--sysconfdir=/etc' '--datadir=/usr/share' '--includedir=/usr/include' '--libdir=/usr/lib64' '--libexecdir=/usr/libexec' '--localstatedir=/var' '--sharedstatedir=/var/lib' '--mandir=/usr/share/man' '--infodir=/usr/share/info' '--sbindir=/usr/lib/frr' '--sysconfdir=/etc/frr' '--localstatedir=/var/run/frr' '--disable-static' '--disable-werror' '--enable-irdp' '--enable-multipath=256' '--enable-vtysh' '--enable-ospfclient' '--enable-ospfapi' '--enable-rtadv' '--enable-ldpd' '--enable-pimd' '--enable-pbrd' '--enable-nhrpd' '--enable-eigrpd' '--enable-babeld' '--enable-vrrpd' '--enable-user=frr' '--enable-group=frr' '--enable-vty-group=frrvty' '--enable-fpm' '--enable-watchfrr' '--disable-bgp-vnc' '--enable-isisd' '--enable-systemd' '--disable-rpki' '--enable-bfdd' 'build_alias=x86_64-redhat-linux-gnu' 'host_alias=x86_64-redhat-linux-gnu' 'PKG_CONFIG_PATH=:/usr/lib64/pkgconfig:/usr/share/pkgconfig'

# yum list installed | grep frr
frr.x86_64                           7.4-01.el8                                 @frr         
frr-pythontools.x86_64               7.5-01.el8                                 @frr         
frr_exporter.x86_64                  0.2.11-1.el8                               @prometheus  
libyang.x86_64                       1.0.109-0                                  @frr-extras  

# cat /etc/frr/frr.conf 
!
! Ansible managed
!
frr defaults traditional
hostname XXXXXXXXX
log file /var/log/frr/frr.log informational
ip forwarding
no ipv6 forwarding
service integrated-vtysh-config
!
! 0.0.0.0
interface ens192
 ip ospf dead-interval 5
 ip ospf hello-interval 1
 ip ospf message-digest-key 1 md5 XXX
!
! 0.0.0.0
interface dummy0
 ip ospf cost 10
!
!
router bgp 64999
 bgp router-id 10.AAA.BBB.72
 bgp log-neighbor-changes
 neighbor iBGP peer-group
 neighbor iBGP remote-as 64999
 neighbor 10.XXX.YYY.9 peer-group iBGP
 neighbor 10.XXX.YYY.9 description fw1
 neighbor 10.XXX.YYY.10 peer-group iBGP
 neighbor 10.XXX.YYY.10 description fw2
 !
 address-family ipv4 unicast
  network 10.XXX.YYY.128/32 route-map REDISTRIBUTE-STATIC
  redistribute kernel route-map REDISTRIBUTE-STATIC
  redistribute connected route-map REDISTRIBUTE-STATIC
  redistribute static route-map REDISTRIBUTE-STATIC
  neighbor iBGP soft-reconfiguration inbound
  neighbor iBGP prefix-list DEFAULT in
  neighbor iBGP prefix-list ALL out
  neighbor iBGP route-map IBGP-IN in
  neighbor iBGP route-map IBGP-OUT out
 exit-address-family
!
router ospf
 ospf router-id 10.AAA.BBB.72
 passive-interface default
 no passive-interface ens192
 network 10.AAA.BBB.64/26 area 0.0.0.0
 network 10.XXX.YYY.128/32 area 0.0.0.0
 area 0.0.0.0 authentication message-digest
!
!
ip prefix-list ALL seq 10 permit 0.0.0.0/0 le 32
ip prefix-list DEFAULT seq 10 permit 0.0.0.0/0
!
bgp community-list standard COMMUNITY_INTERNAL permit 64999:1
bgp community-list standard COMMUNITY_PUBLIC permit 64999:2
bgp community-list standard COMMUNITY_CUSTOMER permit 64999:3
bgp community-list standard COMMUNITY_TRANSIT permit 64999:4
bgp community-list standard COMMUNITY_PEERING permit 64999:5
bgp community-list standard COMMUNITY_POP permit 64999:6
!
route-map REDISTRIBUTE-STATIC deny 10
 match ip address prefix-list DEFAULT
!
route-map REDISTRIBUTE-STATIC permit 20
 match ip address prefix-list ALL
 set community 64999:1
!
route-map IBGP-OUT permit 10
 match community COMMUNITY_INTERNAL
!
route-map IBGP-OUT permit 20
 match community COMMUNITY_PUBLIC
!
route-map IBGP-OUT permit 30
 match community COMMUNITY_CUSTOMER
!
route-map IBGP-OUT permit 40
 match community COMMUNITY_TRANSIT
!
route-map IBGP-OUT permit 50
 match community COMMUNITY_PEERING
!
route-map IBGP-OUT permit 60
 match community COMMUNITY_POP
!
route-map IBGP-OUT deny 90
!
route-map IBGP-IN permit 10
 match community COMMUNITY_INTERNAL
!
route-map IBGP-IN permit 20
 match community COMMUNITY_PUBLIC
!
route-map IBGP-IN permit 30
 match community COMMUNITY_CUSTOMER
!
route-map IBGP-IN permit 40
 match community COMMUNITY_TRANSIT
!
route-map IBGP-IN permit 50
 match community COMMUNITY_PEERING
!
route-map IBGP-IN permit 60
 match community COMMUNITY_POP
!
route-map IBGP-IN deny 90
!
!
line vty
!
```

I can update FRR to 7.5 and FRR exporter to 0.2.12 if you think this will help.

**Comment by tynany** [comment]
Thanks for the detailed information. This has been fixed in release [0.2.13](https://github.com/tynany/frr_exporter/releases/tag/v0.2.13). Please test and confirm you are no longer seeing the issue.

**Comment by karlism** [comment]
Thanks! I've just upgraded to 0.2.13 and it reports status properly after FRR restart, however there is another issue after an upgrade, I will create separate issue for that.
